wallet/src/Pos/Wallet/Web/Methods/Restore.hs

```diff
@@ -55,47 +58,95 @@ import           UnliftIO (MonadUnliftIO)
 initialAccAddrIdxs :: Word32
 initialAccAddrIdxs = firstHardened
 
+mnemonicExists :: Text
+mnemonicExists = "Wallet with that mnemonics already exists"
+
+-- | Creates a wallet. If the wallet with the given passphrase already exists,
+-- then we return 'Left' of the wallet's ID.
 mkWallet
     :: L.MonadWalletLogic ctx m
-    => PassPhrase -> CWalletInit -> Bool -> m (EncryptedSecretKey, CId Wal)
+    => PassPhrase
+    -> CWalletInit
+    -> Bool
+    -> m (Either (CId Wal) (EncryptedSecretKey, CId Wal))
 mkWallet passphrase CWalletInit {..} isReady = do
     let CWalletMeta {..} = cwInitMeta
 
     skey <- genSaveRootKey passphrase cwBackupPhrase
     let cAddr = encToCId skey
 
-    CWallet{..} <- L.createWalletSafe cAddr cwInitMeta isReady
-    -- can't return this result, since balances can change
+    eresult <- fmap Right (L.createWalletSafe cAddr cwInitMeta isReady)
+        `catch` \(e :: WalletError) ->
+            case e of
+                RequestError msg | msg == mnemonicExists ->
```

Comment:
How hard would it be to make `createWalletSafe` actually safe :joy:  ?

Reply:
Very easy! But it'd require touching a lot of the old wallet code that calls this function. This seemed like the easiest way to implement this in the least invasive way possible.